src/main/java/org/testng/internal/Parameters.java

```diff
@@ -69,13 +97,12 @@
     mapping.put(BeforeClass.class.getSimpleName(), ctxTest);
     mapping.put(AfterClass.class.getSimpleName(), ctxTest);
 
-    mapping.put(BeforeMethod.class.getSimpleName(), mParams);
-    mapping.put(AfterMethod.class.getSimpleName(), mParams);
+    mapping.put(BeforeMethod.class.getSimpleName(), beforeAfterMethod);
+    mapping.put(AfterMethod.class.getSimpleName(), beforeAfterMethod);
+    mapping.put(Test.class.getSimpleName(), Arrays.asList(new Class<?>[] {ITestContext.class}));
```

Comment:
This also causes an error because `Collections.singletonList(ITestContext.class)` results in a `List<ITestContext>` but the mapping's type is defined as `Map<String, List<Class<?>>>`.